src/modules/fixEmbedTwitterVideo/fixEmbedTwitterVideo.module.ts

```diff
@@ -0,0 +1,173 @@
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, MessageType } from 'discord.js';
+import { z } from 'zod';
+
+import { createModule } from '../../core/createModule';
+import { resolveCatch } from '../../helpers/resolveCatch.helper';
+
+const FXTwitterResponseSchema = z.object({
+  code: z.number(),
+  message: z.string(),
+  tweet: z.object({
+    media: z
+      .object({
+        videos: z
+          .array(
+            z
+              .object({
+                type: z.enum(['gif', 'video']),
+                url: z.string(),
+              })
+              .optional(),
+          )
+          .optional(),
+      })
+      .optional(),
+  }),
+});
+
+type URLMapping = {
+  pattern: RegExp;
+  replacement: string;
+};
+
+const modulePrefixButtonId = 'fixEmbedTwitterVideo-';
+const deleteBotAnswerPrefixButtonId = modulePrefixButtonId + 'deleteBotAnswer-';
+const deleteBotAnswerButtonId = deleteBotAnswerPrefixButtonId + '$authorIdMessage';
+const ignoreConfirmationButtonId = modulePrefixButtonId + 'ignoreBotButtons';
+
+const twitterUrlMappings: URLMapping[] = [
+  {
+    pattern: /https?:\/\/(mobile\.)?(twitter|x)\.com\/(\S+)\/status\/(\d+)/g,
+    replacement: 'https://fxtwitter.com/$3/status/$4',
+  },
+];
+
+const FXTwitterUrlMappings: URLMapping[] = [
+  {
+    pattern: /https?:\/\/fxtwitter\.com\/(\S+)\/status\/(\d+)/g,
+    replacement: 'https://api.fxtwitter.com/$1/status/$2',
+  },
+];
```

Comment:
not sure why you need two arrays

Reply:
i don't want to iterate on patterns that have no link between them, FXTwitterUrlMappings have a specific purpose and twitterUrlMappings a other